R/estimateAlpha.R

```diff
@@ -0,0 +1,245 @@
+#' 
+#' The function estimates alpha diversity indices optionally using of rarefaction,
+#' then stores results at \code{\link{colData}}.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object.
+#' 
+#' @param assay.type the name of the assay used for
+#'   calculation of the sample-wise estimates.
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity indices
+#'   to be calculated
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices.
+#'   
+#' @param ... optional arguments.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#' rounds.
+#' 
+#' @param rarefaction.depth a \code{double} value as for the minimim size or 
+#' rarefaction.depth. (default: \code{min(colSums(assay(x, "counts")), na.rm = TRUE)})
+#' 
+#' @return \code{x} with additional \code{\link{colData}} named after the index 
+#' used.
+#' 
+#' @examples
+#' 
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' 
+#' # Calculate the default Shannon index with no rarefaction
+#' tse <- estimateAlpha(tse, assay.type = "counts", index = "shannon")
+#' 
+#' # Shows the estimated Shannon index
+#' tse$shannon_diversity
+#'
+#'# Calculate observed richness with 10 rarefaction rounds
+#' tse <- estimateAlpha(tse, assay.type = "counts", index = "observed_richness",
+#' rarefaction.depth=min(colSums(assay(tse, "counts")), na.rm = TRUE), n.iter=10)
+#' 
+#' # Shows the estimated observed richness
+#' tse$observed_richness
+#' 
+#' @importFrom dplyr %>% 
+#' 
+#' @rdname estimateAlpha
+#' @export
+estimateAlpha <- function(x, assay.type = "counts",
+                          index = c("coverage_diversity", "fisher_diversity",
+                                    "faith_diversity", "gini_simpson_diversity",
+                                    "inverse_simpson_diversity",
+                                    "log_modulo_skewness_diversity", "shannon_diversity",
+                                    "absolute_dominance", "dbp_dominance",
+                                    "core_abundance_dominance", "gini_dominance",
+                                    "dmn_dominance", "relative_dominance",
+                                    "simpson_lambda_dominance",
+                                    "camargo_evenness", "pielou_evenness",
+                                    "simpson_evenness", "evar_evenness",
+                                    "bulla_evenness",
+                                    "ace_richness", "chao1_richness", "hill_richness",
+                                    "observed_richness"),
+                          name = index,
+                          ...,
+                          n.iter=10,
+                          rarefaction.depth=max(colSums(assay(x, assay.type)), na.rm = TRUE)){
+    # checks
+    if(is.null(index) & any(!sapply(index, .is_non_empty_string))) {
```

Comment:
You can use argument matching here. I think we have used in other functions. match.arg or something